### ISSUE TITLE
Too many .dylib dependencies

### ISSUE THREAD

**Issue by stefcarlens** [albinoz/ffmpeg-static-OSX#9]
This script does not create a 'real' static FFmpeg build. Compared to the Tessus and Zeranou FFmpeg macOS builds, this one has way too many .dylib dependencies that renders this useless if you want to use it in a sandbox environment.

Here are the dependencies:
/System/Library/Frameworks/OpenAL.framework/Versions/A/OpenAL (compatibility version 1.0.0, current version 1.0.0)
	/System/Library/Frameworks/Foundation.framework/Versions/C/Foundation (compatibility version 300.0.0, current version 1675.129.0)
	/System/Library/Frameworks/AudioToolbox.framework/Versions/A/AudioToolbox (compatibility version 1.0.0, current version 1000.0.0)
	/System/Library/Frameworks/CoreAudio.framework/Versions/A/CoreAudio (compatibility version 1.0.0, current version 1.0.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1281.100.1)
	/System/Library/Frameworks/AVFoundation.framework/Versions/A/AVFoundation (compatibility version 1.0.0, current version 2.0.0)
	/System/Library/Frameworks/CoreVideo.framework/Versions/A/CoreVideo (compatibility version 1.2.0, current version 1.5.0)
	/System/Library/Frameworks/CoreMedia.framework/Versions/A/CoreMedia (compatibility version 1.0.0, current version 1.0.0)
	/System/Library/Frameworks/CoreGraphics.framework/Versions/A/CoreGraphics (compatibility version 64.0.0, current version 1355.13.0)
	/System/Library/Frameworks/OpenGL.framework/Versions/A/OpenGL (compatibility version 1.0.0, current version 1.0.0)
	/usr/lib/libiconv.2.dylib (compatibility version 7.0.0, current version 7.0.0)
	/usr/lib/libexpat.1.dylib (compatibility version 7.0.0, current version 8.0.0)
	/usr/lib/libz.1.dylib (compatibility version 1.0.0, current version 1.2.11)
	/System/Library/Frameworks/CoreImage.framework/Versions/A/CoreImage (compatibility version 1.0.1, current version 5.0.0)
	/System/Library/Frameworks/AppKit.framework/Versions/C/AppKit (compatibility version 45.0.0, current version 1894.40.150)
	/usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 902.1.0)
	/usr/local/opt/xz/lib/liblzma.5.dylib (compatibility version 8.0.0, current version 8.5.0)
	/System/Library/Frameworks/VideoToolbox.framework/Versions/A/VideoToolbox (compatibility version 1.0.0, current version 1.0.0)
	/System/Library/Frameworks/CoreFoundation.framework/Versions/A/CoreFoundation (compatibility version 150.0.0, current version 1675.129.0)
	/System/Library/Frameworks/CoreServices.framework/Versions/A/CoreServices (compatibility version 1.0.0, current version 1069.22.0)
	/System/Library/Frameworks/CoreText.framework/Versions/A/CoreText (compatibility version 1.0.0, current version 1.0.0)
	/usr/lib/libobjc.A.dylib (compatibility version 1.0.0, current version 228.0.0)

Basically the /usr/lib/ dylib will make it not possible to use it when sandboxing or macOS 10.11
Especially these are troublesome:

/usr/lib/libiconv.2.dylib
/usr/lib/libexpat.1.dylib
/usr/local/opt/xz/lib/liblzma.5.dylib

If the script could be take care of those three dependencies (read: make those really static build in), it would be a very good and useful script. ;-)

**Comment by stefcarlens** [comment]
The problem is that some parts are compiled with Homebrew which depends on liblzma.5.dylib. This one is the biggest problem when you want to create a good static FFmpeg build.

Basically Homebrew should be avoided and wget cmake autoconf automake nasm libtool ninja meson should be compiled static as all other parts.

**Comment by rtgoodwin** [comment]
That is how it is done with https://github.com/markus-perl/ffmpeg-build-script which ends up "truly" static. Takes longer of course but it's as portable as it gets.

**Comment by stefcarlens** [comment]
@rtgoodwin : that's a console app which seems to build FFmpeg. Basically, you have no idea what that really does. I only trust a script I can read/check myself. I mean , what reason is there to not use a basic script instead of compiling it. FFmpeg is open source, so should that Markus-perl script be.
It only makes it more suspicious and something everyone should avoid using. Avoid at all cost is my advice.

**Comment by albinoz** [comment]
Thanks for feedback,
Like explain in Read-me, this version of ffmpeg static work between 10.12 and 10.15.
Anyone who want could help for better code to work in older / later mac OS.
I share this code for help some need, and some help to grow this code are welcome ;)

**Comment by shiny-comic** [comment]
@stefcarlens then why are you complain about not static
